modopt/opt/cost.py

```diff
@@ -6,15 +6,17 @@
 
 """
 
+import abc
+
 import numpy as np
 
 from modopt.base.backend import get_array_module
 from modopt.base.types import check_callable
 from modopt.plot.cost_plot import plotCost
 
 
-class costObj(object):
-    """Generic cost function object.
+class AbstractcostObj(abc.ABC):
```

Comment:
For the sake of consistency, perhaps we could call this `CostParent` since it works in a pretty similar way to e.g. `GradParent` or `LinearParent`.  It would nice for the API to be easy to guess by the user.